src/graphics/gl.rs

```diff
@@ -1332,7 +1334,9 @@ impl RenderingBackend for GlContext {
                     unsafe {
                         match attribute.type_ {
                             GL_INT | GL_UNSIGNED_INT | GL_SHORT | GL_UNSIGNED_SHORT
-                            | GL_UNSIGNED_BYTE | GL_BYTE => {
+                            | GL_UNSIGNED_BYTE | GL_BYTE
+                                if !attribute.gl_pass_as_float =>
```

Comment:
QQ: why not something like:
```
                        match attribute.type_ {
                            GL_INT | GL_UNSIGNED_INT | GL_SHORT | GL_UNSIGNED_SHORT
                            | GL_UNSIGNED_BYTE | GL_BYTE
                                if self.has_integer_attributes() =>
                            {
                                glVertexAttribIPointer(
                                    attr_index as GLuint,
                                    attribute.size,
                                    attribute.type_,
                                    attribute.stride,
                                    attribute.offset as *mut _,
                                )
                            }
                            _ => glVertexAttribPointer(
                                attr_index as GLuint,
                                attribute.size,
                                attribute.type_,
                                GL_FALSE as u8,
                                attribute.stride,
                                attribute.offset as *mut _,
                            ),
                        }
```

Reply:
thx, I agree, a lot better!

Reply:
oh, wait, I misread  it. 

`has_integer_attributes` doesn't neccessarily mean we want uvec4 in the shader. 

`has_integer_attributes` means the backend support glVertexAttribIPointer, `gl_pass_as_float` means that the user wants glVertexAttribIPointer to be used.

Reply:
I don't think you d need `gl_pass_as_float`, because the moment it supports integer and the user is passing a byte, it'll convert it. glsl 150 requires opengl 3.2 and above, so that shouldn't affect it. But better test this one.

Reply:
It does not, thats how that whole thing started :) 
glsl100 shaders do not receive glVertexAttribIPointer Byte4 as vec4, even on OpenGL4+